include/fastcdr/xcdr/optional.hpp

```diff
@@ -0,0 +1,400 @@
+    {
+        reset();
+        ::new(&storage_.val_)T(val);
+        storage_.engaged_ = true;
+        return *this;
+    }
+
+    //! Assigns content from an instance of the templated class.
+    optional& operator =(
+            T&& val)
+    {
+        reset();
+        ::new(&storage_.val_)T(std::move(val));
+        storage_.engaged_ = true;
+        return *this;
+    }
+
+    //! Unintialize the optional.
+    optional& operator = (
+            nullopt_t) noexcept
+    {
+        reset();
+        return *this;
+    }
+
+    //! Compares optional values.
+    bool operator ==(
+            const optional& opt_val) const
+    {
+        return opt_val.storage_.engaged_ == storage_.engaged_ &&
+               (storage_.engaged_ ? opt_val.storage_.val_ == storage_.val_ : true);
+    }
+
+    //! Compares optional values.
+    bool operator !=(
+            const optional& opt_val) const
+    {
+        return !operator ==(opt_val);
+    }
+
+    /*!
+     * @brief Accesses the contained value.
+     *
+     * The behavior is undefined if *this does not contain a value.
+     *
+     * @return The contained value.
+     */
+    T& operator *() & noexcept
+    {
+        return storage_.val_;
+    }
+
+    /*!
+     * @brief Accesses the contained value.
+     *
+     * The behavior is undefined if *this does not contain a value.
+     *
+     * @return The contained value.
+     */
+    const T& operator *() const& noexcept
+    {
+        return storage_.val_;
+    }
+
+    /*!
+     * @brief Accesses the contained value.
+     *
+     * The behavior is undefined if *this does not contain a value.
+     *
+     * @return The contained value.
+     */
+    T&& operator *() && noexcept
+    {
+        return std::move(storage_.val_);
+    }
+
+    /*!
+     * @brief Accesses the contained value.
+     *
+     * The behavior is undefined if *this does not contain a value.
+     *
+     * @return The contained value.
+     */
+    const T&& operator *() const&& noexcept
+    {
+        return std::move(storage_.val_);
+    }
+
+    /*!
+     * @brief Accesses the contained value.
+     *
+     * The behavior is undefined if *this does not contain a value.
+     *
+     * @return The contained value.
+     */
+    T* operator ->() noexcept
+    {
+        return std::addressof(storage_.val_);
+    }
+
+    /*!
+     * @brief Accesses the contained value.
+     *
+     * The behavior is undefined if *this does not contain a value.
+     *
+     * @return The contained value.
+     */
+    const T* operator ->() const noexcept
+    {
+        return std::addressof(storage_.val_);
+    }
+
+    //! Checks whether the optional contains a value.
+    explicit operator bool() const noexcept
+    {
+        return storage_.engaged_;
+    }
+
+private:
+
+    detail::optional_storage<T> storage_;
+};       // namespace fastcdr
+
+}        // namespace fastcdr
+}        // namespace eprosima
+
+     #endif //_FASTCDR_XCDR_OPTIONAL_HPP_
```

Comment:
Non-indented. It is funny as the macro to disable uncrustify is introduced in the first part of this file when defining the detail namespace. But the linter is not being passed.

Reply:
Done in e3a6645